src/Stemmer/Catalan.php

```diff
@@ -0,0 +1,302 @@
+        'irien', 'iries', 'irìem', 'irìeu', 'iguem', 'igueu', 'esqui', 'eixin', 'eixis', 'eixen', 'iríem', 'iríeu',
+        'atges', 'issen', 'isses', 'issin', 'issis', 'issiu', 'issim', 'ïssin', 'íssiu', 'íssim', 'ïssis', 'ïguem',
+        'ïgueu', 'ïssen', 'ïsses', 'itzeu', 'itzis', 'ador', 'ents', 'udes', 'eren', 'arán', 'arás', 'aria', 'aràs',
+        'aría', 'arés', 'erán', 'erás', 'ería', 'erau', 'irán', 'irás', 'iría', 'írem', 'íreu', 'aves', 'avem', 'ávem',
+        'àvem', 'àveu', 'áveu', 'aven', 'ares', 'àrem', 'àreu', 'àren', 'areu', 'aren', 'tzar', 'ides', 'ïdes', 'ades',
+        'iera', 'iese', 'aste', 'iste', 'aban', 'aran', 'asen', 'aron', 'abas', 'adas', 'idas', 'aras', 'ases', 'íais',
+        'ados', 'idos', 'amos', 'imos', 'ques', 'iran', 'irem', 'iren', 'ires', 'ireu', 'iria', 'iràs', 'eixi', 'eixo',
+        'isin', 'isis', 'esca', 'isca', 'ïsca', 'ïren', 'ïres', 'ïxen', 'ïxes', 'ixen', 'ixes', 'inin', 'inis', 'ineu',
+        'itza', 'itzi', 'itzo', 'itzà', 'arem', 'ent', 'arà', 'ará', 'ara', 'aré', 'erá', 'eré', 'irá', 'iré', 'íeu',
+        'ies', 'íem', 'ìeu', 'ien', 'uda', 'ava', 'ats', 'ant', 'ïen', 'ams', 'ïes', 'dre', 'eix', 'ïda', 'aba', 'ada',
+        'ida', 'its', 'ids', 'ase', 'ían', 'ado', 'ido', 'ieu', 'ess', 'ass', 'ías', 'áis', 'ira', 'irà', 'irè', 'sis',
+        'sin', 'int', 'isc', 'ïsc', 'ïra', 'ïxo', 'ixo', 'ixa', 'ini', 'itz', 'iïn', 're', 'ie', 'er', 'ia', 'at', 'ut',
+        'au', 'ïm', 'ïu', 'és', 'en', 'es', 'em', 'am', 'ïa', 'it', 'ït', 'ía', 'ad', 'ed', 'id', 'an', 'ió', 'ar',
+        'ir', 'as', 'ii', 'io', 'ià', 'ís', 'ïx', 'ix', 'in', 'às', 'iï', 'iïs', 'í'
+    );
+
+    protected static $residual_suffixes = array(
+        'itz', 'it', 'os', 'eu', 'iu', 'is', 'ir', 'ïn', 'ïs', 'a', 'o', 'á', 'à', 'í', 'ó', 'e', 'é', 'i', 's', 'ì',
+        'ï'
+    );
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stem($word)
+    {
+        // we do ALL in UTF-8
+        if (!UTF8::is_utf8($word)) {
+            throw new \Exception('Word must be in UTF-8');
+        }
+
+        $this->word = UTF8::strtolower($word);
+
+        // Catalan stemmer does not use Rv
+        $this->r1();
+        $this->r2();
+
+        // Step 0: Attached pronoun
+        $this->step0();
+
+        $word = $this->word;
+        // Step 1a: Standard suffix
+        $this->step1a();
+
+        // Step 1b: Verb suffix
+        // Do step 1b if no ending was removed by step 1a.
+        if ($this->word == $word) {
+            $this->step1b();
+        }
+
+        $this->step2();
+        $this->finish();
+
+        return $this->word;
+    }
+
+    /**
+     * Step 0: Attached pronoun
+     *
+     * Search for the longest among the following suffixes
+     * and delete it in R1.
+     */
+
+    private function step0()
+    {
+        if (($position = $this->search(static::$attached_pronoun)) != false) {
```

Comment:
!== 
everywhere
search use strrpos, so it doesn't mind, but good practice for this case

Reply:
Good! I will do.